src/File/Rename.php

```diff
@@ -4,359 +4,183 @@
 
 namespace Laminas\Filter\File;
 
-use Laminas\Filter;
 use Laminas\Filter\Exception;
-use Laminas\Stdlib\ArrayUtils;
-use Traversable;
+use Laminas\Filter\FilterInterface;
 
-use function basename;
-use function count;
+use function array_is_list;
 use function file_exists;
-use function is_array;
-use function is_bool;
+use function fnmatch;
 use function is_dir;
-use function is_scalar;
 use function is_string;
+use function is_writable;
 use function pathinfo;
-use function realpath;
 use function rename;
 use function sprintf;
-use function strlen;
 use function uniqid;
 use function unlink;
 
-use const DIRECTORY_SEPARATOR;
-
 /**
- * @psalm-type Options = array{
- *     file?: array{source?: string, target?: string, overwrite?: bool, randomize?: bool},
- *     ...
+ * @psalm-type OptionsSet = array{
+ *     match?: non-empty-string,
+ *     target_directory?: non-empty-string,
+ *     rename_to?: string,
+ *     overwrite?: bool,
+ *     randomize?: bool
  * }
- * @template TOptions of Options
- * @template-extends Filter\AbstractFilter<TOptions>
+ * @psalm-type Options = OptionsSet|list<OptionsSet>
+ * @psalm-type DefaultedOptionsSet = array{
+ *      match: non-empty-string,
+ *      target_directory: non-empty-string,
+ *      rename_to: string,
+ *      overwrite: bool,
+ *      randomize: bool
+ *  }
+ * @implements FilterInterface<string>
  */
-final class Rename extends Filter\AbstractFilter
+final class Rename implements FilterInterface
 {
-    /**
-     * Internal array of array(source, target, overwrite)
-     *
-     * @var list<array{source: string, target: string, overwrite: bool, randomize: bool}>
-     */
-    protected $files = [];
+    /** @var DefaultedOptionsSet[] */
```

Comment:
Nitpick: I prefer `/** @var list<DefaultedOptionsSet> */` because it is more explicit and it's harder to miss the trailing `[]`